services/app-api/utils/transformations/transformations.ts

```diff
@@ -423,18 +423,24 @@ export const quantitativeQuarters = (
         ? "Second quarter (April 1 - June 30)"
         : "Fourth quarter (October 1 - December 31)";
 
+    //depending on the period, the quarter starts at  Q1 or Q3
+    const currentQuarter = reportPeriod === 1 ? 1 : 3;
     // have to loop twice for both periods
-    for (let quarterNumber = 1; quarterNumber <= 2; quarterNumber += 1) {
+    for (
+      let quarterNumber = currentQuarter;
+      quarterNumber <= currentQuarter + 1;
+      quarterNumber += 1
+    ) {
```

Comment:
Suggestion: Hardcode a tiny array to avoid some math.
```suggestion
    const quartersInPeriod = reportPeriod === 1 ? [1, 2] : [3, 4];
    for ( let quarterNumber of quartersInPeriod) {
```